scanner/src/main/kotlin/ScanResultsCache.kt

```diff
@@ -75,17 +77,44 @@ interface ScanResultsCache {
 
         var stats = CacheStatistics()
 
-        fun configure(config: ArtifactoryCacheConfiguration) {
-            require(config.url.isNotBlank()) {
-                "URL for Artifactory cache is missing."
+        fun configure(config: CacheConfiguration) {
+            config.validate()
+
+            if (config is ArtifactoryCacheConfiguration) {
```

Comment:
Also here we should consider allowing to use multiple storage backends in parallel, simply because we could.

Reply:
This would require more changes in the `ScanResultsCache` which is currently designed to support exactly one cache. Esp. implementing `read` for multiple caches would require some design decisions. Therefore we should do this only in a separate PR, if at all.

Reply:
Agreed.